lfs/scanner.go

```diff
@@ -930,7 +1045,75 @@ func parseLogOutputToPointers(log io.Reader, dir LogDiffDirection,
 	}
 	// Final pointer if in progress
 	finishLastPointer()
+}
+
+// Interface for all types of wrapper around a channel of results and an error channel
+// Implementors will expose a type-specific channel for results
+// Call the Wait() function after processing the results channel to catch any errors
+// that occurred during the async processing
+type ChannelWrapper interface {
+	// Call this after processing results channel to check for async errors
+	Wait() error
+}
+
+// Base implementation of channel wrapper to just deal with errors
+type BaseChannelWrapper struct {
+	errorChan <-chan error
+}
+
+func (w *BaseChannelWrapper) Wait() error {
+
+	var err error
+	for e := range w.errorChan {
+		if err != nil {
+			// Combine in case multiple errors
+			err = fmt.Errorf("%v\n%v", err, e)
```

Comment:
This will lose the backtrace from any scans that return multiple errors. Any panics will show this as the culprit of the bug, and not a scanner that returns a channel wrapper. If it becomes a problem, I could see us using some `MultiError` type that preserves them, and teaching `commands.Panic()` how to handle it.